src/main/java/com/treasuredata/client/TDApiRequest.java

```diff
@@ -98,7 +99,21 @@ public Map<String, String> getQueryParams()
         return queryParams;
     }
 
+    /**
+     * @deprecated Use {@link #getHeaderParamsV2()} instead.
+     * @return
+     */
+    @Deprecated
     public Multimap<String, String> getHeaderParams()
+    {
+        ImmutableMultimap.Builder<String, String> builder = new ImmutableMultimap.Builder<>();
+        for (Map.Entry<String, Collection<String>> e : headerParams.entrySet()) {
+            builder.putAll(e.getKey(), e.getValue());
+        }
+        return builder.build();
+    }
+
+    public Map<String, Collection<String>> getHeaderParamsV2()
```

Comment:
This will be the default method after deprecating getHeaderParams(). How about using a more regular method name like `getAllHeaders()` but a different name from the previous one getHeaderParams()?